GaugeKit/Gauge.swift

```diff
@@ -163,101 +180,112 @@ open class Gauge: UIView {
     var gaugeLayer: CALayer!
 /// Colored layer, depends from scale
     var ringLayer: CAShapeLayer!
+    var ringLayer2:CAShapeLayer!
 /// background for ring layer
     var bgLayer: CAShapeLayer!
 /// ring gradient layer
     var ringGradientLayer: CAGradientLayer!
+/// additional ring gradient layer for circle
+    var ringGradientLayer2: CAGradientLayer!
 /// background gradient
     var bgGradientLayer: CAGradientLayer!
+/// additional background gradient layer for circle
+    var bgGradientLayer2: CAGradientLayer!
 
-    // Animation variables
-    internal var animationTimer: Timer = Timer()
-    internal var animationCompletionBlock: (Bool) -> () = {_ in }
-    
-    func getGauge(_ rotateAngle: Double = 0) -> CAShapeLayer {
+    func getGauge(rotateAngle: Double = 0) -> CAShapeLayer {
         switch type {
-        case .left, .right:
-            return getHalfGauge(rotateAngle)
-        case .circle:
-            return getCircleGauge(rotateAngle)
-        case .line:
-             return getLineGauge(rotateAngle)
+        case .Left, .Right:
+            return getHalfGauge(rotatengle: rotateAngle)
+        case .Circle:
+            return getCircleGauge(rotateAngle: rotateAngle)
+        case .Line:
+            return getLineGauge(rotateAngle: rotateAngle)
         default:
-            return getCircleGauge(rotateAngle)
+            return getCircleGauge(rotateAngle: rotateAngle)
         }
     }
 
     func updateLayerProperties() {
         backgroundColor = UIColor.clear
-        
+
         if (ringLayer != nil) {
-            switch type {
-            case .left, .right:
-                // For Half Gauge, you have to fill 50% of circle and round it wisely
+
+            switch (type) {
+            case .Left, .Right:
+                // For Half gauge you have to fill 50% of circle and round it wisely.
                 let percentage = (rate / 2 / maxValue).truncatingRemainder(dividingBy: 0.5)
                 ringLayer.strokeEnd = (rate >= maxValue ? 0.5 : percentage + ((rate != 0 && percentage == 0) ? 0.5 : 0))
+            case .Circle, .Custom:
+                ringLayer.strokeEnd = rate / maxValue
             default:
                 ringLayer.strokeEnd = rate / maxValue
+
             }
-            
+
             var strokeColor = UIColor.lightGray
+            //TODO: replace pre-defined colors with array of user-defined colors
+            //TODO: and split them proportionally in whole sector
             
-            if !colorsArray.isEmpty {
-                switch colorsArray.count {
-                case 1:
-                    strokeColor = colorsArray.first!
-                case 2:
-                    let percentage: CGFloat = rate / maxValue
-                    strokeColor = (1 - percentage) * colorsArray.first! + percentage * colorsArray.last!
+            //colorsArray = [.red, .green, .blue]
+            
+            if colorsArray {
```

Comment:
It probably will conflict with another solution: https://github.com/skywinder/GaugeKit/pull/68

Did you check it? it looks pretty nice for this thing